### PR TITLE
When Compiling serialize seperately per julia version

### LINKED ISSUE #288: If TimeZone is installed and built on a newer version of Julia that TimeZone version is unusable on older Julia versions
```
julia> using TimeZones # 1.5.1

julia> tz"UTC"
ERROR: LoadError: Cannot read stream serialized with a newer version of Julia.
Got data version 12 > current version 10
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] readheader(::Serialization.Serializer{IOStream}) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/Serialization/src/Serialization.jl:715
 [3] handle_deserialize(::Serialization.Serializer{IOStream}, ::Int32) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/Serialization/src/Serialization.jl:878
 [4] deserialize(::Serialization.Serializer{IOStream}) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/Serialization/src/Serialization.jl:773
 [5] deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/Serialization/src/Serialization.jl:760 [inlined]
 [6] open(::typeof(Serialization.deserialize), ::String, ::Vararg{String,N} where N; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at ./io.jl:325
 [7] open at ./io.jl:323 [inlined]
 [8] (::TimeZones.var"#3#4"{String})() at /Users/kristoffercarlsson/.julia/packages/TimeZones/v0mfN/src/types/timezone.jl:50
 [9] get!(::TimeZones.var"#3#4"{String}, ::Dict{String,Tuple{TimeZone,TimeZones.Class}}, ::String) at ./dict.jl:450
 [10] TimeZone(::String, ::TimeZones.Class) at /Users/kristoffercarlsson/.julia/packages/TimeZones/v0mfN/src/types/timezone.jl:46 (repeats 2 times)
 [11] @tz_str(::LineNumberNode, ::Module, ::Any) at /Users/kristoffercarlsson/.julia/packages/TimeZones/v0mfN/src/types/timezone.jl:86
in expression starting at REPL[2]:1
```

Because I happened to install TimeZones on julia master first.

### PR DESCRIPTION
**Pull request by oxinabox**
Fixes #288 (and its duplicate #309).

I have tested locally that this works to fix that problem.
It's hard to test it in CI since it requires starting multiple julia instances.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name = "TimeZones"
 uuid = "f269a46b-ccf7-5d73-abea-4c690281aa53"
 authors = ["Curtis Vogt <curtis.vogt@gmail.com>"]
-version = "1.3.2"
+version = "1.3.3"
```
Comment (oxinabox) [inline Project.toml:4]: I say this is a bug fix.
This always should have worked but it.

**Comment by oxinabox** [comment]
oops looks like i fixed it on an old version, rebasing and retest now

**Comment by oxinabox** [comment]
confirm still fixed

File: Project.toml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name = "TimeZones"
 uuid = "f269a46b-ccf7-5d73-abea-4c690281aa53"
 authors = ["Curtis Vogt <curtis.vogt@gmail.com>"]
-version = "1.5.2"
+version = "1.5.3"
```
Comment (oxinabox) [inline Project.toml:4]: I say this is a bug-fix. It always should have worked

**Review by fchorney — APPROVED** [review]
LGTM

File: Project.toml (hunk 2 above)
Comment (fchorney) [reply Project.toml:4]: I think that makes sense

**Comment by iamed2** [comment]
Can we make this serialize based on major.minor? Or can serialization change in patch versions?

**Comment by oxinabox** [comment]
It probably can change based on patch versions if there is bug in how serialisation is done that needs changed to the format to fix, which is then backported.
More importantly if you are using Julia nightly/source build it can definitely change based on the pre-release build suffix.

I feel like keeping things simple and just using the whole VERSION is better than trying to be clever.
It makes it marginally slower to add and uses a tiny bit more harddrive.

File: src/tzdata/TZData.jl [hunk 3]
```diff
@@ -16,7 +16,7 @@ end
 
 const ARCHIVE_DIR = joinpath(DEPS_DIR, "tzarchive")
 const TZ_SOURCE_DIR = joinpath(DEPS_DIR, "tzsource")
-const COMPILED_DIR = joinpath(DEPS_DIR, "compiled")
+const COMPILED_DIR = joinpath(DEPS_DIR, string("compiled_julia_v", VERSION))
```
Comment (omus) [inline src/tzdata/TZData.jl:19]: I'd rather use separate sub-directories:
```suggestion
const COMPILED_DIR = joinpath(DEPS_DIR, "compiled", string(VERSION))
```
This requires that we change [this line](https://github.com/JuliaTime/TimeZones.jl/blob/01d025ae6420251e3e075f2dd1eae3f7a9b2521e/src/tzdata/build.jl#L107) to use `mkpath`

**Review by omus — CHANGES_REQUESTED** [review]
Eventually we'll probably change the approach here to use scratch spaces: https://github.com/JuliaTime/TimeZones.jl/pull/295#issuecomment-705302332

File: src/tzdata/TZData.jl (hunk 3 above)
Comment (oxinabox) [reply src/tzdata/TZData.jl:19]: I considered that but then i would also have to change that other line

**Review by omus — APPROVED** [review]
Looks good. Just going to get CI working to ensure everything is working as expected here

**Comment by omus** [comment]
I'm quite sure the benchmark failure is due to the compiled files being in different directories. As a build isn't triggered when running against the old commit then the compiled files are missing according to the base version. As I'm not expecting performance issues from these changes I'll go ahead and merge this.

File: Project.toml (hunk 2 above)
Comment (omus) [reply Project.toml:4]: Definitely is a patch but I like to bump versions separate from PRs for this repo.
```suggestion
version = "1.5.3"
```

**Pull request merged**